src/connector/C#/src/TDengineDriver/TDengineDriver.cs

```diff
@@ -0,0 +1,392 @@
+        public IntPtr is_null;
+        // unused            
+        public int is_unsigned;
+        // unused           
+        public IntPtr error;
+        public Int64 u;
+        public uint allocated;
+    }
+
+
+    [StructLayout(LayoutKind.Sequential)]
+    public struct TAOS_MULTI_BIND
+    {
+        // column type
+        public int buffer_type;
+
+        // array, one or more lines column value
+        public IntPtr buffer;
+
+        //length of element in TAOS_MULTI_BIND.buffer (for binary and nchar it is the longest element's length)
+        public ulong buffer_length;
+
+        //array, actual data length for each value
+        public IntPtr length;
+
+        //array, indicates each column value is null or not
+        public IntPtr is_null;
+
+        // line number, or the values number in buffer 
+        public int num;
+    }
+
+
+    public class TDengine
+    {
+        public const int TSDB_CODE_SUCCESS = 0;
+
+        [DllImport("taos", EntryPoint = "taos_init", CallingConvention = CallingConvention.Cdecl)]
+        static extern public void Init();
+
+        [DllImport("taos", EntryPoint = "taos_cleanup", CallingConvention = CallingConvention.Cdecl)]
+        static extern public void Cleanup();
+
+        [DllImport("taos", EntryPoint = "taos_options", CallingConvention = CallingConvention.Cdecl)]
+        static extern public void Options(int option, string value);
+
+        [DllImport("taos", EntryPoint = "taos_connect", CallingConvention = CallingConvention.Cdecl)]
+        static extern public IntPtr Connect(string ip, string user, string password, string db, short port);
+
+        [DllImport("taos", EntryPoint = "taos_errstr", CallingConvention = CallingConvention.Cdecl)]
+        static extern private IntPtr taos_errstr(IntPtr res);
+        static public string Error(IntPtr res)
+        {
+            IntPtr errPtr = taos_errstr(res);
+            return Marshal.PtrToStringAnsi(errPtr);
+        }
+
+        [DllImport("taos", EntryPoint = "taos_errno", CallingConvention = CallingConvention.Cdecl)]
+        static extern public int ErrorNo(IntPtr res);
+
+        [DllImport("taos", EntryPoint = "taos_query", CallingConvention = CallingConvention.Cdecl)]
+        static extern public IntPtr Query(IntPtr conn, string sqlstr);
+
+        [DllImport("taos", EntryPoint = "taos_affected_rows", CallingConvention = CallingConvention.Cdecl)]
+        static extern public int AffectRows(IntPtr res);
+
+        [DllImport("taos", EntryPoint = "taos_field_count", CallingConvention = CallingConvention.Cdecl)]
+        static extern public int FieldCount(IntPtr res);
+
+        [DllImport("taos", EntryPoint = "taos_fetch_fields", CallingConvention = CallingConvention.Cdecl)]
+        static extern private IntPtr taos_fetch_fields(IntPtr res);
+        static public List<TDengineMeta> FetchFields(IntPtr res)
+        {
+            const int fieldSize = 68;
```

Comment:
why 68?

Reply:
To @sangshuduo :

```c
typedef struct taosField {
  char     name[65];
  uint8_t  type;
  int16_t  bytes;
} TAOS_FIELD;
```

68 is TAOS_FIELD struct size.
offset 65 is to get `type`.
offset 66 is to get `bytes`.

One quetion: should C# use the same definition of TAOS_FIELD or not? @xleili @sangshuduo

Reply:
I'm working to push the engine team to publish more definitions in the header file. Right now I make taosdef.h temporarily public. Could we refer to the definition from the header file instead of the magic number. 
At least, we can define them like, for example, FIELD_LENGTH 65, and FIELD_TYPE  (FIELD_NAME_LENGTH +1), etc.

Reply:
That will be much better to connectors. Adding constants for offsets is good for now.

Reply:
> 
I will define these number, and make the code easier to read. In next submit